app/assets/javascripts/app/views/search_base_view.js

```diff
@@ -0,0 +1,141 @@
+        var nameTokens = Bloodhound.tokenizers.nonword(datum.name);
+        var handleTokens = datum.handle ? Bloodhound.tokenizers.nonword(datum.name) : [];
+        return nameTokens.concat(handleTokens);
+      },
+      queryTokenizer: Bloodhound.tokenizers.whitespace,
+      prefetch: {
+        url: "/contacts.json",
+        transform: this.transformBloodhoundResponse,
+        cache: false
+      },
+      sufficient: 5
+    };
+
+    // The publisher does not define an additionnal source for searchin
+    // This prevents tests from failing when this additionnal source isn't set
+    if(this.searchFormAction !== undefined){
+      bloodhoundConf.remote = {
+        url: this.searchFormAction + ".json?q=%QUERY",
+        wildcard: "%QUERY",
+        transform: this.transformBloodhoundResponse
+      };
+    }
+
+    this.bloodhound = new Bloodhound(bloodhoundConf);
+
+    /**
+     * Custom searching function that let us filter contacts from prefetched Bloodhound results.
+     */
+    this.bloodhound.customSearch = function(query, sync, async){
+      var filterResults = function(datums){
+        return _.filter(datums, function(result){
+          if(result.handle){
+            return !_.contains(self.resultsTofilter, result.handle);
+          }
+        });
+      };
+
+      var _sync = function(datums){
+        var results = filterResults(datums);
+        sync(results);
+      };
+
+      self.bloodhound.search(query, _sync, async);
+    };
+  },
+
+  setupTypeahead: function() {
+    this.typeaheadElement.typeahead({
+          hint: false,
+          highlight: true,
+          minLength: 2
+        },
+        {
+          name: "search",
+          display: "name",
+          limit: 5,
+          source: this.bloodhound.customSearch,
+          templates: {
+            /* jshint camelcase: false */
+            suggestion: HandlebarsTemplates.search_suggestion_tpl
+            /* jshint camelcase: true */
+          }
+        });
+  },
+
+  transformBloodhoundResponse: function(response) {
+    return response.map(function(data){
+      // person
+      if(data.handle){
+        data.person = true;
+        return data;
+      }
+
+      // hashtag
+      return {
+        hashtag: true,
+        name: data.name,
+        url: Routes.tag(data.name.substring(1))
+      };
+    });
+  },
+
+  /**
+   * This bind events to highlight a result when overing it
+   */
+  bindSelectionEvents: function(){
+    var self = this;
+    var onover = function(suggestion){
+      return function(){
+        self.select(suggestion);
+      };
+    };
+
+    this.typeaheadElement.on("typeahead:render", function(){
+      self.$(".tt-menu *").off("mouseover");
+      self.$(".tt-menu .tt-suggestion").each(function(){
+        var $suggestion = $(this);
+        $suggestion.on("mouseover", onover($suggestion));
+        $suggestion.find("*").on("mouseover", onover($suggestion));
+      });
+    });
+  },
```

Comment:
Yeah, but why having two background colors? It's a bit counter intuitive since it seems like the result is selected when it's not. So, when you presse tab or enter, the wrong result will be selected. It looks like a design flaw to me.